test/commands/scanner/run.test.ts

```diff
@@ -948,4 +857,23 @@ describe('scanner:run', function () {
 				expect(ctx.stdout).to.contain('No rule violations found.', 'Should be no violations found in the file.');
 			});
 	});
+
+	// Any commands that specify the --verbose cause subsequent commands to execute as if --verbose was specified.
+	// Put all --verbose commands at the end of this file.
+	describe('Verbose tests must come last. Verbose does not reset', () => {
+		setupCommandTest
+			.command(['scanner:run',
+				'--target', path.join('test', 'code-fixtures', 'apex', 'AbstractPriceRuleEvaluatorTests.cls'),
+				'--format', 'xml',
+				'--verbose'
+			])
+			.it('When the --verbose flag is supplied, info about implicitly run rules is logged', ctx => {
+				// We'll split the output by the <violation> tag, so we can get individual violations.
+				const violations = ctx.stdout.split('<violation');
+				// Before the violations are logged, there should be 16 log runMessages about implicitly included PMD categories.
+				const regex = new RegExp(events.info.categoryImplicitlyRun.replace(/%s/g, '.*'), 'g');
+				const implicitMessages = violations[0].match(regex);
+				expect(implicitMessages || []).to.have.lengthOf(26, `Entries for implicitly added categories from all engines:\n ${JSON.stringify(implicitMessages)}`);
```

Comment:
This increased to 26 because of eslint-lwc. The Implicit message might confuse the users because the Catalog is based on enabled and disabled engines, but the actual execution won't include them. I wonder if we should move the Implicit event into the DefaultRuleManager instead of within the LocalCatalog.